test/functional/wallet_multisig_descriptor_psbt.py

```diff
@@ -120,7 +124,7 @@ def run_test(self):
 
         psbts = []
         self.log.info("Now at least M users check the psbt with decodepsbt and (if OK) signs it with walletprocesspsbt...")
-        for m in range(self.M):
+        for m in random.sample(range(self.M), self.M):
```

Comment:
In 8a2021c0f4f40f6f8a37dd5619ecfa3f38084679 "tests: assert xpub order in sortedmulti descriptors do not matter"

It's not clear to me what this `random.sample` is supposed to do or how it improves the test.

Reply:
All participants are equivalent [as discussed in related review comment](https://github.com/bitcoin/bitcoin/pull/29154#discussion_r1595783449). Maybe it's extra and we shouldn't even bother with `random.sample`. If you'd prefer that lmk and I can remove

Why I included it in first place:
- all participants being equivalent is a critical property of this multisig. I know it's clear behavior of `thresh`. But I went ahead and asserted it in this test. Otherwise as signers decay, without a change in ordering, we are always testing that the same `m` of `N` signers that can recover at 3 of 4, 2 of 4, 1 of 4. If it happened to be that only the first signer can recover at 1 of 4 it would be bad if user of this multisig didn't expect it and that was the key they lost or couldn't recover. The shuffling tries to prove (in a very simple way) that any keys can be the 3 of 4, 2 of 4, or 1 of 4, independent of ordering in the wallet descriptor